sentry/src/main/java/io/sentry/transport/QueuedThreadPoolExecutor.java

```diff
@@ -65,27 +64,27 @@ public QueuedThreadPoolExecutor(
     }
   }
 
+  @SuppressWarnings("FutureReturnValueIgnored")
   @Override
-  protected void beforeExecute(final @NotNull Thread t, final @NotNull Runnable r) {
+  protected void afterExecute(final @NotNull Runnable r, final @Nullable Throwable t) {
     try {
-      super.beforeExecute(t, r);
+      super.afterExecute(r, t);
     } finally {
-      currentlyRunning.incrementAndGet();
+      unfinishedTasksCount.decrement();
     }
   }
 
-  @SuppressWarnings("FutureReturnValueIgnored")
-  @Override
-  protected void afterExecute(final @NotNull Runnable r, final @Nullable Throwable t) {
+  /** Blocks the thread until there are no running tasks. */
+  void waitTillIdle() {
     try {
-      super.afterExecute(r, t);
-    } finally {
-      currentlyRunning.decrementAndGet();
+      unfinishedTasksCount.waitTillZero();
```

Comment:
We need to block until the queue is empty or the timeout elapses. Am I missing something or this would block indefinitely?

Reply:
Good catch. I just missed it. Will add soon.